ui/src/app/subnets/views/SubnetsList.tsx

```diff
@@ -1,13 +1,47 @@
 import React from "react";
 
-const SubnetsList = (): React.ReactElement => (
-  <header className="p-strip--light is-shallow u-no-padding--bottom">
-    <div className="row">
-      <div className="col-medium-4 col-8">
-        <h1 data-testid="section-header-title">Subnets</h1>
-      </div>
-    </div>
-  </header>
-);
+import { ContextualMenu } from "@canonical/react-components";
+
+import Section from "app/base/components/Section";
+import SectionHeader from "app/base/components/SectionHeader";
+import { SubnetForms } from "app/subnets/enum";
+import type { SubnetForm } from "app/subnets/types";
+import FormActions from "app/subnets/views/FormActions";
+
+const SubnetsList = (): React.ReactElement => {
+  const [activeForm, setActiveForm] = React.useState<SubnetForm | undefined>(
+    undefined
+  );
+
+  return (
+    <Section
+      header={
+        <>
+          <SectionHeader
```

Comment:
`SectionHeader` has a `headerContent` prop that is used to render things like forms in the header. It comes with a bit of styling, so using it should mean you can remove the top-level `section` and `div` from `FormActions`.

``` ts
headerContent={
  activeForm ? (
    <FormActions
      activeForm={activeForm}
      setActiveForm={setActiveForm}
    />
  ) : null
}
```

Since the existence check for `activeForm` happens here you could also remove it from `FormActions`.

Reply:
Thanks, didn't realise. I'll use your suggestion.

One thing to note though - I'm not sure it's a good idea to replace the h1 of the page you're on with the active form when the dropdown-activated form is opened. We're still within the context of the same page.

This happens on other pages that use this pattern, e.g. `MAAS/r/controllers`

![image](https://user-images.githubusercontent.com/7452681/149307140-da375842-e4c1-45b1-9791-1afb588e088a.png)

![image](https://user-images.githubusercontent.com/7452681/149307099-36128dc2-c40e-4d15-802f-a7ae7fa75fed.png)

### How I did it on the Subnets page:
![image](https://user-images.githubusercontent.com/7452681/149307536-8da8d72f-d5d7-4648-9b81-02bb098c9936.png)

Reply:
I’m more than happy to go with your suggestion, but you should discuss it with @amylily1011 (and maybe @lyubomir-popov?) who initially proposed the pattern of having a contextual top level header

Reply:
We sort of use the contextual form title instead of the header title because it is within the context of the same page and it is redundant to show the tab name and the tab title when we open the form. I think we should keep the current design (form title on the form header) for now. This is just so we can do an audit on all pages during MAAS IA work to prepare for the next version. 

We also went with the current pattern because we are preparing to transfer form content into a panel, which has a common pattern in Vanilla already. This way, you won't have to keep changing back and forth 😄 .

Reply:
Thanks for pointing that out @petermakowski, this is a recurring problem that we faced for that past few cycles. The problem is people shouldn't see this slide down panel as a page. They should see is as a form and the H1 header provides the interpretation that it's a page, which is not our intention. And you pointing out that it confuses you that it's in a different page confirms that this creates a misconception. 

For this reason, our big goal is to change these forms into panels, which should apply to all products at Canonical.

The H1 header should not reflect the page, it should reflect the form title, because in MAAS the some forms can be triggered from different pages.

Reply:
So in theory the title of the page you're referencing should say "Add subnet", instead of just "Subnet".